packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -147,6 +153,18 @@ export class MediaUpload extends Component {
 							source.types.includes( allowedType )
 					  );
 			} )
+			.filter( ( source ) => {
+				if (
+					allowedTypes.every(
+						( allowedType ) => allowedType === MEDIA_TYPE_AUDIO
+					)
+				) {
+					if ( source.id !== URL_MEDIA_SOURCE ) {
+						return isAudioBlockEnabled === true;
+					}
+				}
```

Comment:
For this change, the idea here is that only when the `media-upload` component is being utilized by the audio block should we remove the media sources if the audio block capability isn't enabled.